packages/neon_framework/packages/account_repository/lib/src/account_repository.dart

```diff
@@ -0,0 +1,325 @@
+        b
+          ..serverURL = Uri.parse(response.server)
+          ..username = response.loginName
+          ..password = response.appPassword;
+      });
+    } on http.ClientException catch (error, stackTrace) {
+      if (error case DynamiteStatusCodeException(statusCode: 404)) {
+        return null;
+      }
+
+      Error.throwWithStackTrace(PollLoginFailure(error), stackTrace);
+    }
+  }
+
+  /// Fetches the information for the account with the given [credentials].
+  ///
+  /// May throw a [FetchAccountFailure].
+  Future<Account> getAccount(Credentials credentials) async {
+    final client = buildClient(
+      credentials: credentials,
+      userAgent: _userAgent,
+      httpClient: _httpClient,
+    );
+
+    try {
+      final response = await client.authentication.users.getCurrentUser();
+
+      return Account((b) {
+        b
+          ..client = client
+          ..credentials.replace(credentials)
+          ..credentials.username = response.body.ocs.data.id;
+      });
+    } on http.ClientException catch (error, stackTrace) {
+      Error.throwWithStackTrace(FetchAccountFailure(error), stackTrace);
+    }
+  }
+
+  /// Logs in the given [account].
+  ///
+  /// It will also set the active account if none is active yet.
+  Future<void> logIn(Account account) async {
+    final value = _accounts.value;
+
+    final active = value.active ?? account.credentials.id;
+    final accounts = value.accounts.rebuild((b) {
+      b[account.credentials.id] = account;
+    });
+
+    _accounts.add((active: active, accounts: accounts));
+    await Future.wait([
+      _storage.saveAccounts(accounts.values.map((e) => e.credentials)),
+      _storage.saveLastAccount(active),
+    ]);
+  }
+
+  /// Logs out the user from the server.
+  ///
+  /// May throw a [DeleteCredentialsFailure].
+  Future<void> logOut(String accountID) async {
+    final value = _accounts.value;
+
+    Account? account;
+    final accounts = value.accounts.rebuild((b) {
+      account = b.remove(accountID);
+    });
+
+    var active = value.active;
+    if (active == accountID) {
+      active = accounts.keys.firstOrNull;
+    }
+
+    await Future.wait([
+      _storage.saveAccounts(accounts.values.map((e) => e.credentials)),
+      _storage.saveLastAccount(active),
+    ]);
+
+    _accounts.add((active: active, accounts: accounts));
+
+    try {
+      await account?.client.authentication.appPassword.deleteAppPassword();
+    } on http.ClientException catch (error, stackTrace) {
+      Error.throwWithStackTrace(DeleteCredentialsFailure(error), stackTrace);
+    }
+  }
+
+  /// Deletes the current user account.
+  ///
+  /// Not yet implemented see: https://github.com/nextcloud/neon/issues/2177
+  Future<void> deleteAccount() async {
+    throw UnimplementedError();
+  }
+
+  /// Sets the active `account` to [accountID].
+  ///
+  /// Throws an [ArgumentError] when the id is not in the accounts map.
+  Future<void> switchAccount(String accountID) async {
```

Comment:
Wouldn't it make sense to pass the entire Account and then use the ID to look it up internally?
Any user of the repository already handles the Account, so having to pass the ID instead feels a little off.